hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/validate/HazelcastSqlValidator.java

```diff
@@ -225,6 +225,21 @@ protected void validateOrderList(SqlSelect select) {
     protected void validateJoin(SqlJoin join, SqlValidatorScope scope) {
         super.validateJoin(join, scope);
 
+        switch (join.getJoinType()) {
+            case LEFT:
+                if (containsStreamingSource(join.getRight())) {
+                    throw newValidationError(join, RESOURCE.streamingSourceInWrongSideOfLeftJoin());
+                }
+                break;
+            case RIGHT:
+                if (containsStreamingSource(join.getLeft())) {
+                    throw newValidationError(join, RESOURCE.streamingSourceInWrongSideOfRightJoin());
+                }
+                break;
+            default:
+                break;
+        }
+
         // the right side of a join must not be a subquery or a VALUES clause
         join.getRight().accept(new SqlBasicVisitor<Void>() {
```

Comment:
Does this case actually matter for swapped Joins? If its a Right join, since we're swapping things, do we want to perform similar check for the `join.getRight()`?